pkg/apb/client.go

```diff
@@ -227,3 +228,21 @@ func OcLogin(log *logging.Logger, args ...string) error {
 	log.Debug(string(output))
 	return nil
 }
+
+// TODO: Instead of putting namespace directly as a parameter, we should create a dictionary
+// of apb_metadata and put context and other variables in it so we don't pollute the user
+// parameter space.
+func createExtraVars(context *Context, parameters *Parameters) (string, error) {
+	var paramsCopy Parameters
+	if parameters != nil && *parameters != nil {
+		paramsCopy = *parameters
+	} else {
+		paramsCopy = make(Parameters)
+	}
+
+	if context != nil {
```

Comment:
What if ```context == nil```? Below we allow this to be empty, but up here we don't set a default value for namespace.